Resources/config/events_deferring.yml

```diff
@@ -3,11 +3,16 @@ parameters:
   event_dispatcher.class: Smartbox\Integration\FrameworkBundle\Tools\EventsDeferring\EventDispatcher
 
 services:
+  deferred_event_dispatcher:
+    class: "%event_dispatcher.class%"
+    arguments:
+      $container: '@service_container'
+
   smartesb.handlers.events:
       class: "%smartesb.handlers.events_deferring.class%"
       calls:
         - [setId, ['smartesb.handlers.events']]
-        - [setEventDispatcher, ['@event_dispatcher']]
```

Comment:
I have been investigating why in Symfony 2.8 this was working in our code and in Symfony 3.4 no and the reason is because in Symfony 2.8 `event_dispatcher.class` was defined as parameter and we were overriding the parameter here in framework and in Symfony 3.4 this parameter doesn't exist anymore and the event_dispatcher is declared directly as service.
This is the code in Symfony 2.8.47
symfony/symfony/src/Symfony/Bundle/FrameworkBundle/Resources/config/services.xml
```
<?xml version="1.0" ?>

<container xmlns="http://symfony.com/schema/dic/services"
    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
    xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">

    <parameters>
        <parameter key="event_dispatcher.class">Symfony\Component\EventDispatcher\ContainerAwareEventDispatcher</parameter>
        <parameter key="http_kernel.class">Symfony\Component\HttpKernel\DependencyInjection\ContainerAwareHttpKernel</parameter>
        <parameter key="filesystem.class">Symfony\Component\Filesystem\Filesystem</parameter>
        <parameter key="cache_warmer.class">Symfony\Component\HttpKernel\CacheWarmer\CacheWarmerAggregate</parameter>
        <parameter key="cache_clearer.class">Symfony\Component\HttpKernel\CacheClearer\ChainCacheClearer</parameter>
        <parameter key="file_locator.class">Symfony\Component\HttpKernel\Config\FileLocator</parameter>
        <parameter key="uri_signer.class">Symfony\Component\HttpKernel\UriSigner</parameter>
        <parameter key="request_stack.class">Symfony\Component\HttpFoundation\RequestStack</parameter>
    </parameters>

    <services>
        <service id="event_dispatcher" class="%event_dispatcher.class%">
            <argument type="service" id="service_container" />
        </service>
```
and this is the code in Symfony 3.4.35
```
<?xml version="1.0" ?>

<container xmlns="http://symfony.com/schema/dic/services"
    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
    xsi:schemaLocation="http://symfony.com/schema/dic/services https://symfony.com/schema/dic/services/services-1.0.xsd">

    <services>
        <defaults public="false" />

        <service id="event_dispatcher" class="Symfony\Component\EventDispatcher\ContainerAwareEventDispatcher" public="true">
            <argument type="service" id="service_container" />
            <tag name="container.hot_path" />
        </service>
        <service id="Symfony\Component\EventDispatcher\EventDispatcherInterface" alias="event_dispatcher" />
       ...
```
To be able to do the change in framework and don't affect to any of the other bundles, we should override the event_dispatcher in the compiler pass to define our class. https://symfony.com/doc/3.4/bundles/override.html#services-configuration

Declare the class names as parameters to allow other bundles override these classes was considered bad practice by symfony. See https://github.com/symfony/symfony/issues/11881

Reply:
*.class parameters removed.